wrapper_writer/scala_parser.py

```diff
@@ -0,0 +1,173 @@
+            sys.exit(1)
+
+    def find_method_regex(self):
+        """
+        This function will find the raw method signature from file to be parsed
+        :return: iterable object with all methods found
+        """
+        retrieve_data = self.read_scala_file()
+        pattern = re.compile("def (\w+)\((.*)\): (\w+)", re.MULTILINE)
+        try:
+            pattern2 = pattern.finditer(retrieve_data)
+        except:
+            print("Nothing In There")
+        return pattern2
+
+    @staticmethod
+    def extract_return_type(raw_res):
+        """
+        This function extracts the return type of a method
+        :param raw_res: A method signature from the file to be parsed
+        :return: return_type: String object of the return type
+        """
+        first, *middle, last = raw_res.split()
+        return_type = last
+        return return_type
+
+    @staticmethod
+    def extract_method_name(raw_res):
+        """
+        The function will use regex to extract the method name from the logic code
+        :param raw_res: A method signature from the file to be parsed
+        :return: clean_func_name: String object of the function name
+        """
+        func_name = r"def (\w+)"
+        func_name_find = re.search(func_name, raw_res)
+        fun_name_raw = func_name_find.group()
+        clean_func_name = fun_name_raw.replace("def ", "")
+        return clean_func_name
+
+    @staticmethod
+    def extract_params(raw_res):
+        """
+        This function will use regex to extract the parameters from the logic code
+        :param raw_res: method signature from the file to be parsed
+        :return new_dict: A dictionary containing the parameters
+        """
+        retrieve_params = r"\((.*)\)"
+        retrieve_params_find = re.search(retrieve_params, raw_res)
+        retrieve_params_raw = retrieve_params_find.group()
+        retrieve_params_clean = retrieve_params_raw[1:-1]
+        dict_by_comma = dict(item.split(":") for item in retrieve_params_clean.split(","))
+        for k, v in dict_by_comma.items():
+            dict_by_comma[k] = v.lstrip()
+            new_dict = {k.lstrip(): v for k, v in dict_by_comma.items()}
+        return new_dict
+
+    def multi_process(self):
+        """
+        This function will process each method found and write it to a yaml file
+        :return:
+        """
+        all_found = self.find_method_regex()
+        matches = tuple(all_found)
+        if not matches:
+            raise Exception("No Methods Found")
+        for i in matches:
+            ig = i.group()
+            base_raw = os.path.basename(self.filename)
+            container_name = os.path.splitext(base_raw)[0]
+            return_type = self.extract_return_type(ig)
+            method_name = self.extract_method_name(ig)
+            params = self.extract_params(ig)
+            data = {container_name: {method_name: {"params": params, "returns": return_type}}}
```

Comment:
I thought this would have been used to create a container class that would then be used to create the yaml string.